support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -33,29 +37,106 @@ import (
 
 // PodReconcilerConfig holds the config passed in when creating the reconciler
 type PodReconcilerConfig struct {
-	Client             client.Client
-	Cluster            string
-	Ctx                context.Context
-	DisabledNamespaces []string
-	Log                logrus.FieldLogger
-	PodLabel           string
-	PodAnnotation      string
-	Scheme             *runtime.Scheme
-	TrustDomain        string
+	Client                client.Client
+	Cluster               string
+	Ctx                   context.Context
+	DisabledNamespaces    []string
+	Log                   logrus.FieldLogger
+	PodLabel              string
+	PodAnnotation         string
+	Scheme                *runtime.Scheme
+	TrustDomain           string
+	IdentityTemplate      string
+	IdentityTemplateLabel string
+	Context               map[string]string
+}
+
+const (
+	PodNameLabel           string = "Name"
+	PodUIDLabel            string = "UID"
+	PodNamespaceLabel      string = "Namespace"
+	PodServiceAccountLabel string = "ServiceAccount"
+	PodHostnameLabel       string = "Hostname"
+	PodNodeNameLabel       string = "NodeName"
+	DefaultSpiffeIDPath    string = "ns/{{.Pod.Namespace}}/sa/{{.Pod.ServiceAccount}}"
+)
+
+// PodInfo is created for every processed Pod and it holds pod specific information
+type PodInfo struct {
+	ServiceAccount string
+	Namespace      string
+	Name           string
+	UID            types.UID
+	Hostname       string
+	NodeName       string
+}
+
+// IdentityMaps is used for forming the text from the templates
+type IdentityMaps struct {
+	Context map[string]string
+	Pod     PodInfo
 }
 
 // PodReconciler holds the runtime configuration and state of this controller
 type PodReconciler struct {
 	client.Client
-	c PodReconcilerConfig
+	c             PodReconcilerConfig
+	identityTempl *template.Template
 }
 
 // NewPodReconciler creates a new PodReconciler object
-func NewPodReconciler(config PodReconcilerConfig) *PodReconciler {
-	return &PodReconciler{
-		Client: config.Client,
-		c:      config,
+func NewPodReconciler(config PodReconcilerConfig) (*PodReconciler, error) {
+	if config.IdentityTemplate == "" && config.PodAnnotation == "" && config.PodLabel == "" {
+		config.IdentityTemplate = DefaultSpiffeIDPath
 	}
+	if config.IdentityTemplate == "" {
+		return &PodReconciler{
+			Client: config.Client,
+			c:      config,
+		}, nil
+	}
+
+	tmpl, err := template.New("IdentityTemplate").Parse(config.IdentityTemplate)
+	if err != nil {
+		config.Log.WithError(err).WithField("identity_template", config.IdentityTemplate).Error("error parsing identity template")
+		return &PodReconciler{}, err
```

Comment:
Wrapping the error will help with traceability back to this code path
```suggestion
		return &PodReconciler{}, fmt.Errorf("parsing identity template: %w", err)
```